File: mayur/StackUsingLinkList.java
import java.util.*;

public class StackUsingLinkList {

    // declaring Node class
    private class Node {
        int data;
        Node next;
    }

    // declare required varibales
    public static Scanner scanner = new Scanner(System.in);
    private Node head;
    private Node newnode;

    // initialising using constructor
    StackUsingLinkList() {
        head = null;
        newnode = null;
    }

    // push function
    public void push() {

        System.out.println("enter element : ");
        int element = scanner.nextInt();

        if (head == null) {
            newnode = new Node();
            newnode.data = element;
            newnode.next = null;
            head = newnode; // allocated space is given to head
        }

        else {
            newnode = new Node();
            newnode.data = element;
            newnode.next = null;

            newnode.next = head;
            head = newnode; // allocated space is given to head
        }

        System.out.println("pushed.");
    }

    // pop function
    public int pop() {

        if (head == null) {
            System.out.println("stack is empty.");
            return -1;
        }
        int element = head.data;
        head = head.next;

        return element;

    }

    // display function
    public void display() {

        Node temp;
        temp = head;
        while (temp.next != null) {
            System.out.println(temp.data);
            temp = temp.next;
        }
        System.out.println(temp.data);
    }

    // driver code
    public static void main(String args[]) {

        int choice;
        int element;

        StackUsingLinkList obj = new StackUsingLinkList();

        do {
            System.out.println("1. push : ");
            System.out.println("2. pop : ");
            System.out.println("3. display : ");

            choice = scanner.nextInt();

            switch (choice) {
            case 1:
                obj.push();
                break;

            case 2:
                element = obj.pop();
                if (element != -1) {
                    System.out.println(element + " popped.");
                }
                break;

            case 3:
                obj.display();
                break;
            }

        } while (choice != 0);
    }
}

File: mayur/stack-using-linklist.cpp
#include<iostream>

using namespace std;

typedef struct Node {
    int data;
    Node *next;
} Node;

Node *head = NULL;
Node *newnode = NULL;

void push() {

    int element;
    
    cout<<"enter element : "<<endl;
    cin>>element;
    
    if(head==NULL) {
        // create newnode
        newnode = new Node();
        newnode->next = NULL;
        newnode->data = element;
        head = newnode;
    }

    else {

        // create newnode
        newnode = new Node();
        newnode->next = NULL;
        newnode->data = element;

        newnode->next = head;
        head = newnode;
        
    }

    cout<<"pushed."<<endl;

}

int pop() {

    if(head==NULL) {
        cout<<"stack is empty"<<endl;
        exit(0);
    }
    Node *temp = NULL;
    temp = head;
    int element = temp->data;
    head = head->next;
    free(temp);
    return element;
}

void display() {

    Node *temp = NULL;
    temp = head;
    while(temp->next!=NULL) {
        cout<<temp->data<<endl;
        temp=temp->next;
    }
    cout<<temp->data<<endl;
}

int main() {

    int choice;
    int element;

    do {

        cout<<"1. push : "<<endl;
        cout<<"2. pop : "<<endl;
        cout<<"3. display : "<<endl;
        cout<<"0. EXIT : "<<endl;

        cin>>choice;

        switch(choice) {

            case 1 :
            push();
            break;

            case 2 : 
            element = pop();
            cout<<element<<"popped"<<endl;
            break;

            case 3 : 
            display();
            break;
        }

    }while(choice!=0);

    return 0;

}
